### PR TITLE
DEN-16 Implement ResourceManagedRuleCreate for Managed Rule

### LINKED ISSUE #26: [DEN-16] Terraform: Add Managed Rule (Profile)

### PR DESCRIPTION
**Pull request by wdawg33**
Closes #26

### CODE REVIEW

File: ec/resources/waf/managed_rule.go [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+							Required:    true,
+						},
+						"target_match": {
+							Type:        schema.TypeString,
+							Description: "Identifies a name or category (e.g., cookie name, query string name, country code, etc.) for the data source defined in the target parameter.",
+							Required:    true,
+						},
+					},
+				},
+				Description: "This array identifies each rule that has been disabled using these properties",
```
Comment (stevenpaz) [inline ec/resources/waf/managed_rule.go:198]: Is this the correct description? Sounds like it's for disabled_rules

File: ec/resources/waf/managed_rule.go [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+	if err != nil {
+		return diag.Errorf("error parsing rule_target_update: %+v", err)
+	}
+
+	managedRuleRequest.RuleTargetUpdates = *ruleTargetUpdates
+
+	// General Settings
+
+	if v, ok := d.GetOk("general_settings"); ok {
+		if generalSettings, err := ConvertInterfaceToGeneralSettings(v); err == nil {
```
Comment (stevenpaz) [inline ec/resources/waf/managed_rule.go:244]: To be consistent with naming, we should name that method `ExpandGeneralSettings` - we gotta go back and do that renaming in other places too
- "Expand": Terraform format => API Model
- "Flatten" API Model => Terraform format

**Review by stevenpaz — APPROVED** [review]
It's good as is, just a couple of nitpicky things I hadn't thought about before. Great work Frank!

File: ec/resources/waf/managed_rule.go (hunk 1 above)
Comment (wdawg33) [reply ec/resources/waf/managed_rule.go]: @stevenpaz great catch, thank you. corrected.

File: ec/resources/waf/managed_rule.go (hunk 2 above)
Comment (wdawg33) [reply ec/resources/waf/managed_rule.go]: @stevenpaz Totally clear now, thanks again. corrected.

**Comment by stevenpaz** [comment]
All good, feel free to merge whenever!

**Pull request merged**